grapher_admin/views.py

```diff
@@ -1434,3 +1547,359 @@ def register_by_invite(request: HttpRequest, code: str):
+            subprocess.check_output('git -C %s rm %s' %
+                                    (current_dataset_folder,
+                                     old_dataset_file_path_escaped),
+                                    shell=True)
+            subprocess.check_output('git -C %s rm %s' %
+                                    (current_dataset_folder,
+                                     old_metadata_file_path_escaped),
+                                    shell=True)
+            delete_previous = True
+            commit_message = 'Updating the dataset: %s'
+        else:
+            commit_message = 'Creating the dataset: %s'
+    else:
+        commit_message = 'Creating the dataset: %s'
+
+    # now saving the dataset to file
+
+    with open(dataset_file_path, 'w', newline='', encoding='utf8') as f:
+        writer = csv.writer(f)
+        writer.writerow(headerlist)
+        current_row = None
+
+        for chunk in entity_chunks:
+            with connection.cursor() as outer_cursor:
+                outer_cursor.execute(sql_query % (id_tuple, ','.join([str(x) for x in chunk])))
+                rows = outer_cursor.fetchall()
+            for row in rows:
+                if not current_row or current_row[0] != row[3] or current_row[1] != row[1]:
+                    if current_row:
+                        current_row[0] = allentities_dict[current_row[0]]
+                        writer.writerow(current_row)
+
+                    current_row = [row[3], row[1]]
+                    for i in range(0, len(varlist)):
+                        current_row.append("")
+
+                theindex = 2 + varlist.index(row[2])
+                current_row[theindex] = row[0]
+
+        current_row[0] = allentities_dict[current_row[0]]
+        writer.writerow(current_row)
+
+    with open(metadata_file_path, 'w', encoding='utf8') as f:
+        json.dump(dataset_meta, f, indent=4)
+
+    changed_files = subprocess.check_output('git add %s %s && '
+                                            'git status --porcelain' %
+                                            (dataset_file_path_escaped,
+                                             metadata_file_path_escaped),
+                                            shell=True, cwd=current_dataset_folder)
+
+    changed_files = changed_files.decode('unicode_escape').splitlines()
+    changed_files_list = []
+    # getting the list of added, modified, deleted files
+    for each in changed_files:
+        if len(each.split('->')) > 1:  # finding the status R (file renamed)
+            filename = each.split('->')[1].strip()
+        else:
+            filename = each[each.find(' ') + 1:].strip()
+        if filename.startswith('"') and filename.endswith('"'):
+            changed_files_list.append(filename[1:-1])
+        else:
+            changed_files_list.append(filename)
+
+    if dataset_filename in changed_files_list or metadata_filename in changed_files_list:
```

Comment:
If we extract the commit info directly from the repo rather than storing it in the database, we should be able to remove this section. We can then just try a `git commit` and if nothing has changed it won't create one.